src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1506,6 +1509,10 @@ private function registerMessengerConfiguration(array $config, ContainerBuilder
         $container->getDefinition('messenger.asynchronous.routing.sender_locator')->replaceArgument(1, $messageToSenderIdsMapping);
 
         foreach ($config['adapters'] as $name => $adapter) {
+            if (0 === strpos($adapter['dsn'], 'amqp://') && !$container->hasDefinition('messenger.adapter.amqp.factory')) {
+                throw new LogicException('The default AMQP adapter is not available. Make sure you have installed and enabled the Serializer component.');
```

Comment:
I think it's a sensible tradeoff. But we need to propose a next step to the user: `(install it by running "composer require symfony/serializer".)`